[FAN YUEH-CHING] iP

e1093051 opened the pull request:
### **Carolyn**

> I must govern the clock not govern by it [-Golda Meir](https://woodenswisdom.com/leadershipIssue.php?issue_id=1244)

This app allows you

- mark and unmark tasks
- add new tasks

All you need to do is

1. Download it
2. **Run it!!**

Features

- [ ] Manage task
- [ ] Manage deadline😎

Compile the program with
`javac Carolyn.java`

The below is the main class
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
};
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

e1093051 committed:
Rename, Greet, Exit

e1093051 committed:
Echo

e1093051 committed:
Add, List

e1093051 committed:
Mark as Done

e1093051 committed:
Fix bugs

e1093051 committed:
ToDos, Events, Deadlines

e1093051 committed:
Automated Text UI Testing

e1093051 committed:
Handle Errors

e1093051 committed:
Delete

e1093051 committed:
Add save function, add two new classes: Storage and TaskList

Data will be lost when app is closed

The bot should remember the list

Add Storage class and make neccessary changes to other classes

Using serializable is the easiest option

e1093051 committed:
Bot understand dates and time

Date and Time treated as String

To provide better flexibility

Parser class is created to handle this, and some related changes

Follows OOP rule

e1093051 committed:
Merge pull request #1 from e1093051/branch-Level-7

Add save function, add two new classes: Storage and TaskList

e1093051 committed:
Merge branch 'master' into branch-Level-8

e1093051 committed:
Merge pull request #2 from e1093051/branch-Level-8

Bot understand dates and time

e1093051 committed:
Fix bug from merging

Some bugs come from merging

Fix the bugs

Change two files

The easiest way to do so

e1093051 committed:
Add new class UI and fix some bugs

Some parts not follow OOP rules, some bugs were found

For better readability and good SE practice

Add a new class UI, fix some bugs

UI stuff can be handled by a UI object

e1093051 committed:
Add files to packages

Code files created are not in a package

Make the program organized

Create a package carolyn under java

The java file is the source root

e1093051 committed:
Merge pull request #3 from e1093051/add-gradle-support

Merge Add gradle support

e1093051 committed:
Create and run two test cases with gradle

JUnit tests not available

Need JUnit to do developer testing

Set up the environment and create tests

Gradle provides an easy to to conduct the test

e1093051 committed:
Add Javadoc

No javadoc for methods

Enhance code readability and maintainability.

Add Javadoc for methods

Facilitates easier onboarding and ensures consistent documentation standards.

e1093051 committed:
Amend code to follow coding standard

e1093051 committed:
Add feature find

No such feature before

Want to find tasks based on keyword in description

Change Parser to identify this new command, TaskList to do finding and Carolyn to use these methods

Follow OOP rule and make the least change of code

e1093051 committed:
Merge pull request #4 from e1093051/branch-A-JavaDoc

Add Javadoc

e1093051 committed:
Merge branch 'master' into branch-A-CodingStandard

e1093051 committed:
Merge pull request #5 from e1093051/branch-A-CodingStandard

Amend code to follow coding standard

e1093051 committed:
Merge branch 'master' into branch-Level-9

e1093051 committed:
Merge pull request #6 from e1093051/branch-Level-9

Add feature find

e1093051 committed:
fix minor bugs from merging

SuspectBlue commented on src/main/java/carolyn/Ui.java:
```diff
@@ -0,0 +1,129 @@
+        this.scanner = new Scanner(System.in);
+    }
+  
+    String INDENT = "    ";
```
I believe you intent for INDENT to be a constant. In that case, you should add the static and final keywords to it.

SuspectBlue commented on src/main/java/carolyn/CarolynException.java:
Consider using more specific exception types instead of a single general exception for more precise error handling.

SuspectBlue commented on src/main/java/carolyn/Task.java:
Consider organizing similar classes into one folder to split them up in packages. For example you can place the Task.java, Event.java, Deadline.java, Todo.java and TaskList.java all in a task folder.

SuspectBlue reviewed (commented):
LGTM!
Just some minor nits.🙂

e1093051 committed:
Adds GUI

Uses text-based UI

GUI is more user-friendly

Adds several classes and makes some changes to existing files

Follows JavaFX

e1093051 committed:
Merge pull request #7 from e1093051/branch-Level-10

Adds GUI

e1093051 committed:
Add assertions

No way to check the validity of assumption.

Ensure certain important assumptions are correct.

Add several assertions.

Assertions are for checking potential errors of programmer.

e1093051 committed:
Improve code quality

Some code doesn't follow code standard.

The code is not of high quality.

Delete unused methods, follow naming rules.

Improve code readability.

e1093051 committed:
Set up serial version

e1093051 committed:
Merge pull request #8 from e1093051/branch-A-Assertions

Add assertions

e1093051 committed:
Merge branch 'master' into branch-A-CodeQuality

e1093051 committed:
Merge pull request #9 from e1093051/branch-A-CodeQuality

Two branches not sync.

Master branch needs to follow code quality standard.

Merge branch-A-CodeQuality to master.

Brach-A-CodeQuality follows quality standard.

e1093051 committed:
Add function tag and fix minor errors

e1093051 committed:
Merge pull request #10 from e1093051/branch-C-Tagging

Add function tag and fix minor errors

e1093051 committed:
Fix coding style issues

e1093051 committed:
Add user guide.

e1093051 committed:
Split getResponse method to three shorter methods

e1093051 committed:
Fix minor errors in printed strings

e1093051 committed:
Merge branch 'master' of https://github.com/e1093051/ip

* 'master' of https://github.com/e1093051/ip:
  Add user guide.

e1093051 committed:
Fix typo in user guide

e1093051 committed:
Fix wrong example command in user guide

e1093051 committed:
Split parse to shorter methods

e1093051 committed:
Merge branch 'master' of https://github.com/e1093051/ip

* 'master' of https://github.com/e1093051/ip:
  Fix wrong example command in user guide

e1093051 committed:
Add product photo

e1093051 committed:
Show bot name in title bar

e1093051 committed:
Update product photo

e1093051 committed:
Update user guide product photo

e1093051 committed:
Add AI.md

e1093051 committed:
Merge branch 'master' of https://github.com/e1093051/ip

* 'master' of https://github.com/e1093051/ip:
  Update user guide product photo

e1093051 committed:
Add comments and error throwing and handling

e1093051 committed:
Add AI reference

e1093051 committed:
Update user guide

e1093051 committed:
Update README.md

e1093051 committed:
Delete docs/README.md
